src/lib/eth.ts

```diff
@@ -56,3 +61,12 @@ export function isDapperProvider() {
 export function isValidChainId(chainId: string | number) {
   return Object.values(ChainId).includes(Number(chainId))
 }
+
+export function getChainIdByNetwork(network: Network) {
+  const connectedChainId = getConnectedProviderChainId()
+  if (!connectedChainId) {
+    throw new Error('Could not get connected provider chain id')
+  }
+  const config = getChainConfiguration(connectedChainId)
```

Comment:
This method doesn't fail if the chain configuration isn't found for that chain id. What's the desired output of this function?

Reply:
`getChainConfiguration(chainId: ChainId): ChainConfiguration` takes a `chainId` and always return a `ChainConfiguration` for all the values of the enum `ChainId`, so it can't return `null` or throw. 

This helper is used to get the `chainId` of a given `Network`. For instance, on the Account dapp we have several parts where we have the `Network` but we don't know the `chainId`, with this helper we can do:

```tsx
<ChainButton chainId={getChainIdByNetwork(network)}>
```